apps/condo/domains/marketplace/access/MarketItemFile.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Generated by `createschema marketplace.MarketItemFile 'marketItem:Relationship:MarketItem:CASCADE; file:File;'`
+ */
+
+const { get } = require('lodash')
+
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+const { getById } = require('@open-condo/keystone/schema')
+
+const { canReadMarketItems } = require('@condo/domains/marketplace/access/MarketItem')
+const { checkPermissionInUserOrganizationOrRelatedOrganization } = require('@condo/domains/organization/utils/accessSchema')
+const { RESIDENT } = require('@condo/domains/user/constants/common')
+const { STAFF } = require('@condo/domains/user/constants/common')
+
+const canReadMarketItemFiles = canReadMarketItems
+
+async function canManageMarketItemFiles ({ authentication: { item: user }, originalInput, operation, itemId }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin || user.isSupport) return true
+
+    if (user.type === RESIDENT) return false
+
+    if (user.type === STAFF) {
+        if (operation === 'create') {
+            const marketItemId = get(originalInput, ['marketItem', 'connect', 'id'], null)
+
+            if (marketItemId) {
+                const marketItem = await getById('MarketItem', marketItemId)
+                const organizationId = get(marketItem, 'organization', null)
+
+                return await checkPermissionInUserOrganizationOrRelatedOrganization(user.id, organizationId, 'canManageMarketItems')
+            }
+
+            return false
+        }
+
+        const marketItemFile = await getById('MarketItemFile', itemId)
+        if (!marketItemFile) return false
+
+        const { createdBy, organization } = marketItemFile
+        if (!organization) return createdBy === user.id
```

Comment:
Everyone with permission should edit files.
```suggestion
        const {organization } = marketItemFile
        if (!organization) return false
```